### PR TITLE
Added  Delete Option for Set-AzVmssStorageProfile and Add-AzVmssDataD…

### PR DESCRIPTION
**Pull request by pselugar**
…isk cmdlets

## Description

Please find the Design request here 
- https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1176
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ Specifies the type of virtual machine image (VMImage) offer.
 To obtain an image offer, use the Get-AzVMImageOffer cmdlet.
 
 ```yaml
-Type: System.String
+Type: String
```
Comment (isra-fel) [inline src/Compute/Compute/help/Set-AzVmssStorageProfile.md:154]: Be sure to follow [Updating all markdown files in a module](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module). These types should have full names.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs [hunk 2]
```diff
@@ -69,6 +69,11 @@ public partial class AddAzureRmVmssDataDiskCommand : Microsoft.Azure.Commands.Re
             ValueFromPipelineByPropertyName = true)]
         public string CreateOption { get; set; }
 
+        [Parameter(
+            Mandatory =false,
```
Comment (isra-fel) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:73]: nit: space
```suggestion
            Mandatory = false,
```

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs [hunk 3]
```diff
@@ -330,6 +335,27 @@ private void Run()
                 this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.OsDisk.CreateOption = this.OsDiskCreateOption;
             }
 
+            //DiskDeletion
+            if (this.IsParameterBound( c => c.OsDiskDeletionOption))
```
Comment (isra-fel) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:339]: nit: space
```suggestion
            if (this.IsParameterBound(c => c.OsDiskDeletionOption))
```

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md [hunk 4]
```diff
@@ -266,11 +266,30 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -OsDiskDeletionOption
+"Specifies whether OS disk should be deleted or detached upon VMSS Flex deletion (This feature is available for VMSS with Flexible OrchestrationMode only)
```
Comment (isra-fel) [inline src/Compute/Compute/help/Set-AzVmssStorageProfile.md:270]:
```suggestion
Specifies whether OS disk should be deleted or detached upon VMSS Flex deletion (This feature is available for VMSS with Flexible OrchestrationMode only)
```

File: src/Compute/Compute/help/Add-AzVmssDataDisk.md [hunk 5]
```diff
@@ -81,6 +81,25 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -DeleteOption
+"Specifies whether data disk should be deleted or detached upon VMSS Flex deletion (This feature is available for VMSS with Flexible OrchestrationMode only)
```
Comment (isra-fel) [inline src/Compute/Compute/help/Add-AzVmssDataDisk.md:85]:
```suggestion
Specifies whether data disk should be deleted or detached upon VMSS Flex deletion (This feature is available for VMSS with Flexible OrchestrationMode only)
```

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md [hunk 6]
```diff
@@ -266,11 +266,30 @@ Accept pipeline input: True (ByPropertyName)
...
+
+Accepted Values 
+Delete - If this value is used, the OS disk is deleted when the VMSS Flex VM is deleted.
+Detach -  If this value is used, the OS disk is retained after VMSS Flex VM is deleted.
```
Comment (isra-fel) [inline src/Compute/Compute/help/Set-AzVmssStorageProfile.md:274]:
```suggestion
Detach - If this value is used, the OS disk is retained after VMSS Flex VM is deleted.
```

File: src/Compute/Compute/help/Add-AzVmssDataDisk.md [hunk 7]
```diff
@@ -81,6 +81,25 @@ Accept pipeline input: False
...
+
+Accepted Values 
+Delete - If this value is used, the data disk is deleted when the VMSS Flex VM is deleted.
+Detach -  If this value is used, the data disk is retained after VMSS Flex VM is deleted.
```
Comment (isra-fel) [inline src/Compute/Compute/help/Add-AzVmssDataDisk.md:89]:
```suggestion
Detach - If this value is used, the data disk is retained after VMSS Flex VM is deleted.
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
I find it a little confusing that one parameter is called `*Delete*Option` while another is `OsDisk*Deletion*Option`. Is thare any particular reason they are different?

**Comment by grizzlytheodore** [comment]
could you add powershell design to the description?

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs [hunk 8]
```diff
@@ -87,6 +87,11 @@ public partial class SetAzureRmVmssStorageProfileCommand : Microsoft.Azure.Comma
             ValueFromPipelineByPropertyName = true)]
         public string OsDiskCreateOption { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:92]: I want to push for good practice at this early stage, to add "HelpMessage:" here. 
Look for other examples and the message itself, I usually get from swagger

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs [hunk 9]
```diff
@@ -69,6 +69,11 @@ public partial class AddAzureRmVmssDataDiskCommand : Microsoft.Azure.Commands.Re
...
+            ValueFromPipelineByPropertyName = true)]
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:74]: help message here also

File: src/Compute/Compute/help/Add-AzVmssDataDisk.md (hunk 5 above)
Comment (grizzlytheodore) [inline src/Compute/Compute/help/Add-AzVmssDataDisk.md:85]: where did you get this writing?

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs [hunk 10]
```diff
@@ -69,6 +69,11 @@ public partial class AddAzureRmVmssDataDiskCommand : Microsoft.Azure.Commands.Re
...
+        public string DeleteOption { get; set; }
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:75]: use argument completer to give accepted values. look for examples

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs [hunk 11]
```diff
@@ -87,6 +87,11 @@ public partial class SetAzureRmVmssStorageProfileCommand : Microsoft.Azure.Comma
...
+        public string OsDiskDeletionOption { get; set; }
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:93]: use argument completer

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/help/Set-AzVmssStorageProfile.md:154]: did these changes come from use the script Priyam?

Comment (pselugar) [reply src/Compute/Compute/help/Set-AzVmssStorageProfile.md:154]: Yes they were added as part of running the script

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs (hunk 2 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:73]: Good catch! will remove

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs (hunk 8 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:92]: Got it, will do thanks

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md (hunk 1 above)
Comment (pselugar) [reply src/Compute/Compute/help/Set-AzVmssStorageProfile.md:154]: What should I do, should I revert them? I did not make these changes manually.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs:339]: Done

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md (hunk 4 above)
Comment (pselugar) [reply src/Compute/Compute/help/Set-AzVmssStorageProfile.md:270]: Done made the change

File: src/Compute/Compute/help/Add-AzVmssDataDisk.md (hunk 7 above)
Comment (pselugar) [reply src/Compute/Compute/help/Add-AzVmssDataDisk.md:89]: Done

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md (hunk 6 above)
Comment (pselugar) [reply src/Compute/Compute/help/Set-AzVmssStorageProfile.md:274]: Done

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs (hunk 9 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:74]: Done thanks

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs (hunk 10 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssDataDiskCommand.cs:75]: Done, thanks

**Comment by grizzlytheodore** [comment]
please update description for this PR and add powershell design link

**Comment by pselugar** [comment]
> could you add powershell design to the description?

Added powershell design to the description

**Comment by pselugar** [comment]
> please update description for this PR and add powershell design link

Updated

File: src/Compute/Compute/ChangeLog.md [hunk 12]
```diff
@@ -20,6 +20,7 @@
 
 -->
 ## Upcoming Release
+* Added Disk Delete Optional parameter to the `Set-AzVmssStorageProfile` (OS Disk) and `Add-AzVmssDataDisk` (Data Disk)
```
Comment (grizzlytheodore) [inline src/Compute/Compute/ChangeLog.md:23]: let's write the actual parameter like the example below

Comment (grizzlytheodore) [reply src/Compute/Compute/ChangeLog.md:23]: line 26/27 in this file

**Comment by grizzlytheodore** [comment]
LGTM @isra-fel  could you review/merge?

**Comment by isra-fel** [comment]
Again, my question is about the inconsistency of parameter names: `*Delete*Option` vs. `OsDisk*Deletion*Option`. They may cause some confusion for customers. Which one is more formal? Can you use one word for both parameters?

Also please resolve the conflicts.

File: src/Compute/Compute/ChangeLog.md (hunk 12 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:23]: Good catch will do

**Comment by isra-fel** [comment]
@pselugar Hi, I [checked the design](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1176#:~:text=**%5B%5B%2DOSDiskDeleteOption%20%3CString%3E%5D%5D**) and it read `OsDiskDeleteOption`. So please correct the name of this parameter. Don't forget to refresh the documents.
Thanks.

**Comment by pselugar** [comment]
> @pselugar Hi, I [checked the design](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1176#:~:text=**%5B%5B%2DOSDiskDeleteOption%20%3CString%3E%5D%5D**) and it read `OsDiskDeleteOption` (rather than "Deletion"). So please correct the name of this parameter. Don't forget to refresh the documents. Thanks.

Thanks for the catch. Made the changes and generated corresponding .md files.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**